maturity-level-one/tests/Codit.IntegrationTest/PlayersTest.cs

```diff
@@ -0,0 +1,227 @@
+using System;
+using Xunit;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.AspNetCore.Hosting;
+using System.Net.Http;
+using System.Net;
+using System.Threading.Tasks;
+using Codit.LevelOne.Models;
+using Newtonsoft.Json;
+using System.Text;
+using System.Net.Http.Headers;
+using Microsoft.AspNetCore.JsonPatch;
+
+namespace Codit.IntegrationTest
+{
+    public class PlayersTest
+    {
+        private readonly HttpClient _httpClient;
+        public PlayersTest()
+        {
+            if (_httpClient != null) return;
+            var srv = new TestServer(new WebHostBuilder()
+                .UseEnvironment("Development")
+                .UseStartup<Codit.LevelOne.Startup>());
+
+            _httpClient = srv.CreateClient();
+        }
+
+        [Theory]
+        [InlineData("GET")]
+        public async Task GetPlayers_Ok_TestAsync(string httpMethod)
+        {
+            var request = new HttpRequestMessage(new HttpMethod(httpMethod), "/world-cup/v1/players");
+            var response = await _httpClient.SendAsync(request);
+            response.EnsureSuccessStatusCode();
```

Comment:
I think asserting on the status code is enough because if you test non happy statuses it will fail with an exception without an indication what the returned status code was

Reply:
This applies for all tests

Reply:
do you want to remove  response.EnsureSuccessStatusCode(); for all the happy flows?

Reply:
For all tests actually as the asserts should verify this and check with the expected status code

Reply:
if i add the EnsureSuccessStatusCode(); to the 4XX 5XX tests i won't be able to check the status codes because it will throw the error before the assert and i want to assert the code.

Reply:
response.StatusCode does not throw an exception so I'm not sure what your problem is here?

Reply:
the problem ... it's my english.
Again, if i understood correctly u want to remove the EnsureSuccessStatusCode() which completely make sense. So i think we agree but probably for a better understanding you should speak and write italian ;)

Reply:
lol